### DISCUSSION TITLE
UICollection scroll issue after migration 2.5.3 to 4.2.0.

### DISCUSSION THREAD

**Discussion by Hydralo** [airbnb/lottie-ios#2079] [General]
Hello!

It may be my fault but I am facing an issue after migrating from 2.5.3 to 4.2.0.
I have a collection of images (animated and static) which is scrolling on vertical axis. I've recorded a video from a physical device (iPhone 13), when I am scrolling up and down frame rate dramatically lowers and you can see some animation freezes.
This happens only after migration to 4.2.0 and never happens before.

I've read some docs, announces and closed issues, so I've figured out that the problem can be connected with new rendering engine and json decoding.
So I've tried to build `LottieAnimationView` with configuration `LottieConfiguration(renderingEngine: .mainThread, decodingStrategy: .legacyCodable)` and got no positive changes.
While trying to debug I used Profiler and all traces points to Lottie calls, but I don't find it useful enough to figure out what  actually causes this behaviour.

You can find assets examples which I used in test in the attachment. Also I've attached small video with collection behavior and for some purpose and screenshot from profiler (don't think it's too useful but let it be).

https://github.com/airbnb/lottie-ios/assets/20144920/ff69bf2e-2783-4c04-a3b9-0a6a8fff3e01
<img width="1359" alt="Screenshot 2023-06-19 at 13 48 36" src="https://github.com/airbnb/lottie-ios/assets/20144920/4a6c6342-864e-4beb-bba8-31a16e625b5f">
[assets.zip](https://github.com/airbnb/lottie-ios/files/11788404/assets.zip)

**Comment by Hydralo** [comment]
Moreover, I've builded a standalone project with only collection view and locally placed assets to test cells scrolling with some random set of animations and with the set, I've previously attached. The result is the same.
[CollectionTest.zip](https://github.com/airbnb/lottie-ios/files/11801812/CollectionTest.zip)

**Comment by YSAUM** [comment]
Hello !

I'm experiencing the same symptoms (I have two lotties in a collectionView cell that only animate when clicked) after upgrading from 3.5.0 to 4.2.0. My scrolling performance has dropped drastically, with occasional freezes. 

Here's the scroll hitch rate of my application to illustrate my point and I also noticed a marked increase in battery consumption.
(4.9.12 is the version updated with Lottie 4.2.0, 4.9.11 and prior to version 3.5.0)

<img width="900" alt="Capture d’écran 2023-06-27 à 13 36 41" src="https://github.com/airbnb/lottie-ios/assets/85493547/c3d63219-c572-4f97-bb99-0891c421aace">
<img width="900" alt="Capture d’écran 2023-06-27 à 13 35 49" src="https://github.com/airbnb/lottie-ios/assets/85493547/e2acc41a-6f68-4630-be1a-496344c8c6ab">
